src/main/java/org/folio/innreach/repository/InnReachTransactionRepository.java

```diff
@@ -0,0 +1,14 @@
+package org.folio.innreach.repository;
+
+import org.folio.innreach.domain.entity.InnReachTransaction;
+import org.folio.innreach.domain.entity.TransactionHold;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public interface InnReachTransactionRepository extends JpaRepository<InnReachTransaction, UUID> {
+  @Query(name = InnReachTransaction.FETCH_TRANSACTION_HOLD_NAME)
+  Optional<TransactionHold> findTransactionHoldById(UUID id);
```

Comment:
is there any use case to load transaction hold without transaction itself?

Reply:
No, there's no such use case, added this method just for test.

Reply:
Let's then remove it and add a method that will load transaction with all it's related entities instead

Reply:
Also, I saw `Hibernate.unproxy` usage in tests, let's avoid this too by using a method which loads everything needed